### ISSUE TITLE
Please upgrade to ugorji/go/codec v1.1.1 containing 32-bit fixes (when available)

### ISSUE THREAD

**Issue by anthonyfok** [coreos/etcd#9447]
Hello,

The upcoming release of ugorji/go/codec contains fixes for 32-bit architectures:

* ugorji/go#232: Failed to use CBOR encoding when compili(n)g for GOARCH=386
* [#890886 - golang-github-ugorji-go-codec FTBFS on 32bit: constants overflow int - Debian Bug report logs](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=890886)

Also, ugorji/go/codec now supports time.Time natively as per latest MessagePack specification:

* ugorji/go#224 and ugorji/go#225
* ugorji/go@8badb25: codecgen: support time.Time natively, and handle omitEmpty for structs
* https://github.com/msgpack/msgpack/blob/master/spec.md#timestamp-extension-type

though the following new errors may appear due to the change in codecgen generated code:

```
client/keys.generated.go:474:23: cannot use x.Expirat
ion (type *time.Time) as type time.Time in argument to r.encDriver.EncodeTime
client/keys.generated.go:494:23: cannot use x.Expiration (type *time.Time) as type time.Time in argument to r.encDriver.EncodeTime
```

And here is my suggested patch to client/keys.go and client/keys_test.go to match the changes to time.Time support in the latest ugorji/go/codec:

``` patch
--- a/client/keys.go
+++ b/client/keys.go
@@ -301,7 +301,7 @@
 	ModifiedIndex uint64 `json:"modifiedIndex"`
 
 	// Expiration is the server side expiration time of the key.
-	Expiration *time.Time `json:"expiration,omitempty"`
+	Expiration time.Time `json:"expiration,omitempty"`
 
 	// TTL is the time to live of the key in second.
 	TTL int64 `json:"ttl,omitempty"`
--- a/client/keys_test.go
+++ b/client/keys_test.go
@@ -718,7 +718,7 @@
 					ModifiedIndex: 12,
 					CreatedIndex:  10,
 					TTL:           10,
-					Expiration:    &expiration,
+					Expiration:    expiration,
 				},
 				PrevNode: nil,
 			},
```

Many thanks!

P.S. In case you were wondering, we, the Debian Go Packaging Team, were trying to fix this bug that we encounter on Debian (and Ubuntu) [#890939 - etcd FTBFS - Debian Bug report logs](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=890939) where vendoring is not used.

**Comment by gyuho** [comment]
@anthonyfok Is `*time.Time` type change really necessary? I worry about backward compatibilities.

**Comment by gyuho** [comment]
Oh, doesn't even compile. So that change has to happen...

**Comment by gyuho** [comment]
@ugorji Is changing `*time.Time` to `time.Time` necessary? Is there anyway to make it work without changing it? I tried latest `codecgen` but our `client` cannot be compiled with `*time.Time` field.

**Comment by ugorji** [comment]
I'm away celebrating Easter, and can look at this as early as April 5.

**Comment by ugorji** [comment]
@gyuho please try again and let me know if this is fully fixed on tip. If it is, I will do some sanity testing and tag a github.com/ugorji/go/codec@v1.1.1 release this weekend.

**Comment by ugorji** [comment]
@gyuho please try again. Should be fixed now with ugorji/go@b4c50a2

**Comment by gyuho** [comment]
@ugorji Could you release v1.1.1? We regenerated with https://github.com/ugorji/go/commit/f3cacc17c85ecb7f1b6a9e373ee85d1480919868, and it works (see https://github.com/coreos/etcd/pull/9494).

Thanks!
